src/client/public/scripts/components/dashboard-row.ts

```diff
@@ -62,20 +64,55 @@ export function genericDashboardRowEventTemplate(data: DashboardRowEventData):
 
 export function genericDashboardRowTemplate(
     data: DashboardRowData,
-    isNewlyActionable: boolean,
     extraHeaderData?: TemplateResult[],
     extraEvents?: TemplateResult[]): TemplateResult {
   const additionalClasses: string[] = data.classes || [];
   const rowClasses = ['dashboard-row', ...additionalClasses];
-  if (isNewlyActionable) {
-    rowClasses.push('is-newly-actionable');
-  }
   if (extraEvents && extraEvents.length > 0) {
     rowClasses.push('has-events');
   }
 
   const status = data.status;
 
+  const hasActivityClasses = ['dashboard-row-status__has-activity'];
+  if (data.hasNewActivity) {
+    hasActivityClasses.push('enabled');
+  }
+
+  async function handleRowClick(event: Event) {
+    let hasActivityElement: HTMLElement|null = null;
+
+    let testNode = event.target as HTMLElement;
+    while (testNode) {
```

Comment:
I would prefer:
```
while (testNode && !testnode.classList.contains('dashboard-row-link') {
  testNode = testNode.parentNode;
}

if (testNode) {
   testNode.querySelector(..);
}
```

This will avoid the break.